src/index.js

```diff
@@ -149,14 +152,29 @@ async function bundleMDX(
 
   const bundled = await esbuild.build(buildOptions)
 
-  const decoder = new StringDecoder('utf8')
+  if(bundled.outputFiles){
```

Comment:
I wanted to use `buildOptions.write === false` here but that causes a type error when reading `bundled.outputFiles` as it can't be inffered from checking the build options that it isn't undefined. If this was a `.ts` file I'd just throw a ! on it as we __know__ that if write is false `bundled.outputFiles` must exist.